firmware/applications/SignBuddy/crc.c

```diff
@@ -0,0 +1,38 @@
+#include "crc.h"
+
+#include "board.h"
+
+void init_crc(void)
+{
+  LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_CRC);
+}
+
+uint32_t compute_crc(uint8_t *buffer, uint8_t buffer_size)
+{
+  register uint32_t data = 0;
+  register uint32_t index = 0;
+
+  for (index = 0; index < (buffer_size / 4); index++) {
+    data =
+      (uint32_t) ((buffer[4 * index + 3] << 24) | (buffer[4 * index + 2] << 16) | (buffer[4 * index + 1] << 8) |
+                  buffer[4 * index]);
+    LL_CRC_FeedData32(CRC, data);
```

Comment:
is there a performance advantage of using `LL_CRC_FeedData32` over `LL_CRC_FeedData8` 4 times? If not, this code could be way simpler

Reply:
there is